ext/fg/js/popup-proxy.js

```diff
@@ -51,92 +55,109 @@ class PopupProxy {
 
     // Public functions
 
+    async prepare() {
+        const {id} = await this._invokeHostApi('getOrCreatePopup', {id: this._id, parentId: this._parentId});
+        this._id = id;
+    }
+
     isProxy() {
         return true;
     }
 
+    broadcastRootPopupInformation() {
```

Comment:
IMO this implementation should be put into frontend, and frontend should be responsible for checking if it is the root popup. The sending of messages to the entire frame seems like something outside the scope of popups.

Reply:
https://github.com/FooSoft/yomichan/pull/417/commits/67c5a8c2fb3fdb0050b7ec448ac8a030c0c9d1fb